ion/type_test.go

```diff
@@ -19,3 +19,32 @@ func TestIntSizeToString(t *testing.T) {
 		}
 	}
 }
+
+func TestIsScalar(t *testing.T) {
+	if NoType.IsScalar() {
+		t.Errorf("Expected IsScalar() to return false for type NoType")
+	}
+	for i := NullType; i <= BlobType; i++ {
```

Comment:
Might be safer to define the expected behavior here more explicitly, as a guard against someone reordering the Type enum values?

Reply:
Done.